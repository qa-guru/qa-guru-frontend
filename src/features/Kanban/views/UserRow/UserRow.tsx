import React from "react";
import { Stack, Typography } from "@mui/material";
import { IUserRow } from "./UserRow.types";
import AvatarCustom from "../../../../shared/components/AvatarCustom";

const UserRow: React.FC<IUserRow> = ({ icon: Icon, user }) => {
  const fullName = `${user?.firstName} ${user?.lastName}`;

  return (
    <Stack direction="row" spacing={1} alignItems="center">
      <Icon />
      <AvatarCustom
        fullName={fullName}
        width={26}
        height={26}
        variant="subtitle2"
      />
      <Stack direction="row">
        <Typography variant="body2">{fullName}</Typography>
      </Stack>
    </Stack>
  );
};

export default UserRow;
