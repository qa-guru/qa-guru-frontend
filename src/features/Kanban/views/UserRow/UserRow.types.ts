import React from "react";
import { UserDto } from "../../../../api/graphql/generated/graphql";

export interface IUserRow {
  icon: React.ComponentType;
  user: UserDto;
}
