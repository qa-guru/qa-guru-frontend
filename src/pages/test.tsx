import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
