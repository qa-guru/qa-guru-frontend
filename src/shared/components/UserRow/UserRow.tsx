import React from "react";
import { Box, Stack, Typography } from "@mui/material";
import { format, parseISO } from "date-fns";
import { IUserRow } from "./UserRow.types";
import AvatarCustom from "../AvatarCustom";

const UserRow: React.FC<IUserRow> = (props) => {
  const {
    icon: Icon,
    user,
    date,
    width,
    height,
    variant = "subtitle1",
  } = props;
  const fullName = `${user?.firstName} ${user?.lastName}`;

  return (
    <Stack direction="row" spacing={1.7} alignItems="center">
      {Icon && <Icon />}
      <AvatarCustom
        fullName={fullName}
        width={width}
        height={height}
        variant="subtitle2"
      />
      <Box>
        <Typography variant={variant}>{fullName}</Typography>
        {date && (
          <Stack direction="row" justifyContent="space-between">
            <Typography variant="subtitle2">
              {format(parseISO(date), "dd.MM.yyyy '|' HH:mm")}
            </Typography>
          </Stack>
        )}
      </Box>
    </Stack>
  );
};

export default UserRow;
