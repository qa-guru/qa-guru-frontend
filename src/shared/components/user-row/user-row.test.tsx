import { MemoryRouter } from "react-router-dom";
import { render } from "test/utils-test";
import { studentHomeWorkDto } from "shared/mocks/student-homework-dto.mock";
import UserRow from "./user-row";

describe("UserRow", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UserRow user={studentHomeWorkDto} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
