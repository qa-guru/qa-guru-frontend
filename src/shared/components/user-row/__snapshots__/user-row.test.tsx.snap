// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserRow > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-czl7p-MuiStack-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1d0j7ho-MuiAvatar-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-4ozz3r-MuiTypography-root"
      >
        uu
      </h6>
    </div>
    <div
      class="MuiBox-root css-189yone"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-v3tflm-MuiTypography-root"
      >
        undefined undefined
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
